pytest_fixtures/reporting_fixtures.py

```diff
@@ -0,0 +1,58 @@
+import datetime
+
+import pytest
+from _pytest.junitxml import xml_key
+from xdist import get_xdist_worker_id
+
+from robottelo.config import settings
+from robottelo.decorators import setting_is_set
+
+
+FMT_XUNIT_TIME = "%Y-%m-%dT%H:%M:%S"
+
+
+def pytest_report_header(config):
+    """Called when pytest session starts"""
+    messages = []
+
+    shared_function_enabled = 'OFF'
+    scope = ''
+    storage = 'file'
+    if setting_is_set('shared_function'):
```

Comment:
is there a case in which this wouldn't be set? I believe the facade layer runs through all settings before this point.

Reply:
This is unchanged from the existing functionality, I just moved the function.  I can update it, but would prefer to keep the scope related to how the unit test operates and whether the hook runs on xdist workers.